onnxscript/rewriter/__init__.py

```diff
@@ -42,19 +41,33 @@ def call(self, model: ir.Model) -> ir.passes.PassResult:
 
 
 def rewrite(
-    model: ModelProtoOrIr,
-    pattern_rewrite_rules: Union[Sequence[PatternRewriteRule], pattern.RewriteRuleSet] = (),
-) -> ModelProtoOrIr:
+    model: _ModelProtoOrIr,
+    pattern_rewrite_rules: Union[Sequence[pattern.RewriteRule], pattern.RewriteRuleSet] = (),
+) -> _ModelProtoOrIr:
+    """Rewrite the model using the provided pattern rewrite rules.
+
+    Unused nodes, functions, and opsets will be removed after the rewrite.
+
+    Args:
+        model: The model to be rewritten. Can be an ONNX ModelProto or an ir.Model.
+        pattern_rewrite_rules: A sequence of pattern rewrite rules or a RewriteRuleSet.
+            If not provided or empty, only clean up passes will be applied.
+
+    Returns:
+        The rewritten model, either as an ONNX ModelProto or an ir.Model.
+    """
     if isinstance(model, onnx.ModelProto):
         model_ir = ir.serde.deserialize_model(model)
         proto = True
     else:
         model_ir = model
         proto = False
 
+    rewrite_or_empty = [RewritePass(pattern_rewrite_rules)] if pattern_rewrite_rules else ()
+
     rewrite_pass = ir.passes.PassManager(
         (
-            RewritePass(pattern_rewrite_rules),
+            *rewrite_or_empty,
```

Comment:
The input to the PassManager constructor allows nested lists? Sounds a bit unusual ...

Reply:
This should be a flat list? I’m trying to unpack here. In any case will change to default rewrite set

Reply:
Sorry, I missed the * ... you will still need to have the special handling in case user explicitly passes an empty list (though it will hopefully not happen much)

Reply:
Done